functions.php

```diff
@@ -120,40 +120,77 @@ function panel_login_fail( $username ) {
     if ( !empty($referrer) && !strstr($referrer,'wp-login') && !strstr($referrer,'wp-admin') ) {
         $referrer = explode('?', $referrer);
         parse_str($referrer[1], $referrer_args);
-        unset($referrer_args['idilogout']); 	//unset old idilogout arg if set
-        $referrer_args['idilogin'] = "failed"; 	//let theme know there are errors
+        unset($referrer_args['idilogout']);     //unset old idilogout arg if set
+        $referrer_args['idilogin'] = "failed";     //let theme know there are errors
 
         wp_redirect( $referrer[0]."?".http_build_query($referrer_args) );
         exit;
     }
 }
 
 function idi_display_twitter_feed($twitter_un) {
-    require_once("wp-content/plugins/twitteroauth/twitteroauth/twitteroauth.php");
+    /*
+    We use the oAuth Twitter plugin to help us get the twitter information.
+    https://wordpress.org/plugins/oauth-twitter-feed-for-developers/
+    */
     $num_tweets = 1;
-    $consumerkey = "luK78NyRjDEmMVhi6sgIw";
-    $consumersecret = "E6bY0ShFmtibIqWU0oHokCVZKYtPEvZcNyACBPzYqo";
-    $accesstoken = "123905660-3gtwAKtHHrPjGwa1PmAqXD8FKKKQY2C1ORB8dpyE";
-    $accesstokensecret = "kWuqrTk8IOHVSQdEK9dFfy337VjQv9P44lYKVfq8Fv7Ln";
-    $connection = new TwitterOAuth($consumerkey, $consumersecret, $accesstoken, $accesstokensecret);
-    $tweets = array_filter($connection->get("https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name=".$twitter_un."&count=".$num_tweets));
-
-	echo '<div class="idi-box idi-highlight-box twitter-feed-group">
-				<div class="idi-box-text">
-					<a class="twitter-follow-button" rel="external nofollow" href="http://twitter.com/'.$twitter_un.'" title="Follow @'.$twitter_un.'">@'.$twitter_un.'</a>
-					<ul>
-	';
-	if(!empty($tweets)){
-	        foreach($tweets as $tweet) {
-                echo '<li class="tweet">'.$tweet->text.
-                '<div class="tweet-date">'.substr($tweet->created_at, 0, 16).'</div></li>';
+    $tweets = getTweets($num_tweets, $twitter_un);
+
+    echo '<div class="idi-box idi-highlight-box twitter-feed-group">
+                <div class="idi-box-text">
+                    <a class="twitter-follow-button" rel="external nofollow" href="http://twitter.com/'.$twitter_un.'" title="Follow @'.$twitter_un.'">@'.$twitter_un.'</a>
+                    <ul>';
+    foreach($tweets as $tweet){
+        if($tweet['text']){
+            $the_tweet = $tweet['text'];
+            /*
+            Format the Twitter information for output. This is a modified version of the source script. It's been changed to match our classnames and layout.
+            Source: https://github.com/stormuk/storm-twitter-for-wordpress/wiki/Example-code-to-layout-tweets
+
+
+            Convert Tweet Entities within the Tweet text to link to their URLs on Twitter. */
+
+            // Convert User_mentions to link to user profiles.
+            if(is_array($tweet['entities']['user_mentions'])){
```

Comment:
These 3 section of code for converting `user_mentions`/`hashtags`/`entities` are quite similar. Please refactor the code to use a common function for it. This function could have 4 parameters:

1. `the_tweet`;
2. The field name. The value in each case would be `user_mentions`, `hashtags`, `entities`;
3. `pattern`. The first argument of `preg_replace()`;
4. `replacement`. The second argument of `preg_replace()`.